app/scripts/metamask-controller.js

```diff
@@ -3290,9 +3410,17 @@ export default class MetamaskController extends EventEmitter {
         lastBalance = await this.getBalance(addedAccountAddress, ethQuery);
       }
 
+      const internalAccounts = this.accountsController.listAccounts();
+
       // remove extra zero balance account potentially created from seeking ahead
       if (accounts.length > 1 && lastBalance === '0x0') {
-        await this.removeAccount(accounts[accounts.length - 1]);
+        await this.removeAccount(
+          internalAccounts.find(
+            (account) =>
+              account.address.toLowerCase() ===
+              accounts[accounts.length - 1].toLowerCase(),
+          ).address,
+        );
```

Comment:
`find` can return `undefined`, do we need to check for it?

``` suggestion
        const internalAccount = internalAccounts.find(
          (account) =>
            account.address.toLowerCase() ===
            accounts[accounts.length - 1].toLowerCase(),
        );
        if (internalAccount) {
          await this.removeAccount(internalAccount.address);
        }
```

Or throw an error?

``` suggestion
        const internalAccount = internalAccounts.find(
          (account) =>
            account.address.toLowerCase() ===
            accounts[accounts.length - 1].toLowerCase(),
        );
        if (internalAccount) {
          await this.removeAccount(internalAccount.address);
        } else {
          throw new Error("Could not find account");
        }
```

Reply:
I agree that is better to handle the `undefined` case but in this case it should be ok to not throw an error. Solved Throwing an error sounds appropriate for this case. Done in [957a205](https://github.com/MetaMask/metamask-extension/pull/21437/commits/957a205fe9e0a800c0b8bbff8ac3e314456ace88)